sno/commit.py

```diff
@@ -147,5 +193,29 @@ def get_commit_message(repo, wc_changes):
     # - whitespace at start/end
     # - comment lines
     # - blank lines surrounding comment lines
-    message = re.sub(r"^\n*#.*\n", "", message.strip(), flags=re.MULTILINE)
+    message = re.sub(r"^\n*#.*\n", "", message, flags=re.MULTILINE)
     return message.strip()
+
+
+def commit_output_to_text(jdict):
+    jdict = jdict["sno.commit/v1"]
+    branch = jdict["branch"]
+    commit = jdict["abbrevCommit"]
+    message = jdict["message"].replace("\n", " ")
+    diff = diff_status_to_text(jdict["changes"])
+    datetime = to_long_local_iso8601(from_short_iso8601(jdict["commitTime"]))
+    return f"[{branch} {commit}] {message}\n{diff}\n  Date: {datetime}"
+
+
+def to_short_iso8601(datetime):
+    """Returns a string like: 2020-03-26T09:10:11Z"""
+    return datetime.strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
isn't this just `f"{datetime.isoformat()}Z"`

Reply:
datetime is a terrible variable name though

Reply:
I think it is datetime.isoformat().replace("+00:00", "Z") -- is that better?

I was a bit uncompromising on my vision - I started with an idea of how I wanted timestamps to look in JSON, and then forced poor datetime.datetime to comply.

Reply:
> I think it is datetime.isoformat().replace("+00:00", "Z") -- is that better?

You're writing it, any further and I'm totally bikeshedding.

> I was a bit uncompromising on my vision

👍 As long as we're consistent everywhere, that's fine.